### PR TITLE
Updates dev container and adds doc for getting started with GitHub Codespaces

### PR DESCRIPTION
**Pull request by samruddhikhandale**
Reference - https://github.com/mastodon/mastodon/issues/23346

Hi 👋 

I am Samruddhi, a Software Engineer for GitHub, working with the [GitHub Codespaces](https://github.com/features/codespaces) team, and a maintainer for the [dev containers](https://github.com/devcontainers) org 👩‍💻

I would like to contribute improvements to the existing dev container and update documentation to improve the GitHub Codespaces experience.

image

You can test these changes with Codespaces by simply clicking on the `Code` icon on top right side of this PR ☝️ 

Thanks,
Samruddhi

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,39 +1,31 @@
...
-  // Comment out to connect as root instead. More info: https://aka.ms/vscode-remote/containers/non-root.
-  "remoteUser": "vscode"
+		// Configure tool-specific properties.
+		"customizations": {
+				// Configure properties specific to VS Code.
+				"vscode": {
+						// Set *default* container specific settings.json values on container create.
+						"settings": {},
+						// Add the IDs of extensions you want installed when the container is created.
+						"extensions": ["EditorConfig.EditorConfig", "webben.browserslist"]
```
Comment (nightpool) [inline .devcontainer/devcontainer.json:28]: Why remove the ruby and eslint extensions?

File: README.md [hunk 2]
```diff
@@ -62,6 +62,16 @@ Mastodon acts as an OAuth2 provider, so 3rd party apps can use the REST and Stre
 
 ## Deployment
 
+### Getting Started with GitHub Codespaces
```
Comment (nightpool) [inline README.md:65]: I'm uncertain whether we want to mention this in the README, but if we do, it certainly shouldn't be in the `Deployment` section. Certainly nobody is going to be using Codespaces to run production mastodon servers.

Comment (nightpool) [reply README.md:65]: Probably it would make the most sense to mention this below the Vagrant section below.

File: .devcontainer/Dockerfile [hunk 3]
```diff
@@ -1,16 +1,14 @@
-# [Choice] Ruby version (use -bullseye variants on local arm64/Apple Silicon): 3, 3.1, 3.0, 2, 2.7, 2.6, 3-bullseye, 3.1-bullseye, 3.0-bullseye, 2-bullseye, 2.7-bullseye, 2.6-bullseye, 3-buster, 3.1-buster, 3.0-buster, 2-buster, 2.7-buster, 2.6-buster
-ARG VARIANT=3.1-bullseye
-FROM mcr.microsoft.com/vscode/devcontainers/ruby:${VARIANT}
+# For details, see https://github.com/devcontainers/images/tree/main/src/ruby
+FROM mcr.microsoft.com/devcontainers/ruby:0-3.0-bullseye
```
Comment (nightpool) [inline .devcontainer/Dockerfile:2]: Why downgrade the Ruby version?

File: .devcontainer/devcontainer.json [hunk 4]
```diff
@@ -1,39 +1,31 @@
+// For more details, see https://aka.ms/devcontainer.json.
 {
-  "name": "Mastodon",
-  "dockerComposeFile": "docker-compose.yml",
-  "service": "app",
-  "workspaceFolder": "/mastodon",
+		"name": "Mastodon",
+		"dockerComposeFile": "docker-compose.yml",
+		"service": "app",
```
Comment (nightpool) [inline .devcontainer/devcontainer.json:5]: Why change the indentation throughout this file? The new indentation doesn't match our `.editorconfig` (https://github.com/mastodon/mastodon/blob/main/.editorconfig) and makes the diff harder to review.

File: .devcontainer/devcontainer.json [hunk 5]
```diff
@@ -1,39 +1,31 @@
...
 
-  // Use 'forwardPorts' to make a list of ports inside the container available locally.
-  // This can be used to network with other containers or the host.
-  "forwardPorts": [3000, 4000],
-
-  // Use 'postCreateCommand' to run commands after the container is created.
-  "postCreateCommand": ".devcontainer/post-create.sh",
-
-  // Comment out to connect as root instead. More info: https://aka.ms/vscode-remote/containers/non-root.
-  "remoteUser": "vscode"
```
Comment (nightpool) [inline .devcontainer/devcontainer.json:38]: Why remove this property?

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (samruddhikhandale) [reply .devcontainer/devcontainer.json:28]: I am trying to cleanup duplication, the ruby image used by this dev container already adds the [rebornix.Ruby here](https://github.com/devcontainers/images/blob/main/src/ruby/.devcontainer/devcontainer.json#L27). Also, it adds the Node Feature which adds the [eslint extension here](https://github.com/devcontainers/features/blob/main/src/node/devcontainer-feature.json#L40). 

The supporting tools using the dev container CLI [metadata logic](https://containers.dev/implementors/spec/#merge-logic) merges these customizations.

File: README.md (hunk 2 above)
Comment (samruddhikhandale) [reply README.md]: Makes sense, moved.

File: .devcontainer/Dockerfile (hunk 3 above)
Comment (samruddhikhandale) [reply .devcontainer/Dockerfile]: Updated to use 3.2

File: .devcontainer/devcontainer.json (hunk 4 above)
Comment (samruddhikhandale) [reply .devcontainer/devcontainer.json]: Reverted the tab indentation changes 👍

File: .devcontainer/devcontainer.json [hunk 6]
```diff
@@ -1,39 +1,37 @@
+// For more details, see https://aka.ms/devcontainer.json.
 {
-  "name": "Mastodon",
-  "dockerComposeFile": "docker-compose.yml",
-  "service": "app",
-  "workspaceFolder": "/mastodon",
+	"name": "Mastodon",
```
Comment (nschonni) [inline .devcontainer/devcontainer.json:3]: Running `yarn format` should address the whitespace changes

Comment (samruddhikhandale) [reply .devcontainer/devcontainer.json]: Thanks for the pointer, updated!

**Review by Gargron — APPROVED** [review]

**Pull request merged**